airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/integration/test_report_based_streams.py

```diff
@@ -0,0 +1,266 @@
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_compressed_report_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name, params in STREAMS.items():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(_check_report_status_request(_REPORT_ID).build(), _check_report_status_response(stream_name))
+                http_mocker.get(
+                    _get_document_download_url_request(_REPORT_DOCUMENT_ID).build(),_get_document_download_url_response(compressed=True)
+                )
+
+                # a workaround to pass compressed document to the mocked response
+                document_request = _download_document_request(_DOCUMENT_DOWNLOAD_URL).build()
+                document_response = _download_document_response(stream_name, data_format=params.get("data_format", "csv"), compressed=True)
+                document_request_matcher = HttpRequestMatcher(document_request, minimum_number_of_expected_match=1)
+                http_mocker._matchers.append(document_request_matcher)
+
+                http_mocker._mocker.get(
+                    requests_mock.ANY,
+                    additional_matcher=http_mocker._matches_wrapper(document_request_matcher),
+                    response_list=[{"content": document_response.body, "status_code": document_response.status_code}],
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_report_access_forbidden_when_read_then_no_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name in STREAMS.keys():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response(status_code=403))
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 0
+
+    @HttpMocker()
+    def test_given_report_status_cancelled_when_read_then_no_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name in STREAMS.keys():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(
+                    _check_report_status_request(_REPORT_ID).build(),
+                    _check_report_status_response(stream_name, processing_status=ReportProcessingStatus.cancelled)
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 0
```

Comment:
Should there be an error to the users in that case?

Reply:
In the current implementation, there's a warning log message. Added this assertion to the test